archive_repo.py

```diff
@@ -0,0 +1,667 @@
+                pageInfo {
+                    endCursor
+                    hasNextPage
+                }
+                totalCount
+            }
+        }
+    }
+    rateLimit {
+        remaining
+        resetAt
+    }
+}
+"""
+    + gql_Comment_Fields
+)
+
+gql_PR_Review_Query = (
+    """
+query($id: ID!, $cursor: String!){
+    node(id: $id) {
+        ...on PullRequest {
+            reviews(first:100, after:$cursor) {
+                nodes { ...reviewFields }
+                pageInfo {
+                    endCursor
+                    hasNextPage
+                }
+                totalCount
+            }
+        }
+    }
+    rateLimit {
+        remaining
+        resetAt
+    }
+}
+"""
+    + gql_Review_Fields
+)
+
+gql_PR_ReviewComments_Query = (
+    """
+query($id: ID!, $cursor: String!){
+    node(id: $id) {
+        ...on PullRequestReview {
+            comments(first: 50, after:$cursor) {
+                nodes {
+                    originalPosition
+                    ...commentFields
+                }
+                pageInfo {
+                    endCursor
+                    hasNextPage
+                }
+                totalCount
+            }
+        }
+    }
+    rateLimit {
+        remaining
+        resetAt
+    }
+}
+"""
+    + gql_Comment_Fields
+)
+
+##########################
+## Function definitions ##
+##########################
+
+
+last_request_limit = 5000
+
+
+def submit_query(query, variables, display):
+    global last_request_limit
+
+    url = "https://api.github.com/graphql"
+
+    bodyjson = {"query": re.sub("\s+", " ", query).strip()}
+    if variables:
+        bodyjson["variables"] = variables
+    body = json.dumps(bodyjson)
+
+    if not args.quiet:
+        output = f"Submitting query for {display} with "
+        output += str(variables) if variables else "no parameters"
+        print(output)
+
+    for attempt in range(3):
+        try:
+            response = s.post(url, body)
+            response.raise_for_status()
+            break
+        except requests.exceptions.HTTPError as e:
+            # TODO:  This might not be what rate-limit looks like in GraphQL;
+            #   I haven't managed to hit the rate limit yet.
+            if (
+                e.response.status_code == "403"
+                and "x-ratelimit-reset" in e.response.headers.keys()
+            ):
+                # We're rate-limited; STALL
+                reset = int(e.response.headers["x-ratelimit-reset"])
+                time_to_sleep = reset - datetime.datetime.now().timestamp() + 1
+                print(
+                    "GitHub API rate-limited; waiting for"
+                    + str(time_to_sleep)
+                    + "seconds"
+                )
+                time.sleep(time_to_sleep)
+                continue
+            else:
+                print(e.request.url)
+                print(e.request.headers)
+                print(e.response.headers)
```

Comment:
These might be worth making conditional on `not args.quiet`, i.e., use the log function.
```suggestion
                log(e.request.url)
                log(e.request.headers)
                log(e.response.headers)
```